src/meta/src/serving/mod.rs

```diff
@@ -58,7 +60,8 @@ impl ServingVnodeMapping {
                     FragmentDistributionType::Unspecified => unreachable!(),
                     FragmentDistributionType::Single => Some(1),
                     FragmentDistributionType::Hash => None,
-                };
+                }
+                .or_else(|| max_serving_parallelism.map(|p| p as usize));
```

Comment:
IIUC, the current strategy is using 1 parallelism for single distribution and max_serving_parallelism for hash distribution.

Reply:
Yes.